### ISSUE TITLE
Add a user_id column to the modx_sessions table

### ISSUE THREAD

**Issue by muzzwood** [modxcms/revolution#16624]
### Summary
Add a user_id column to the `modx_sessions` table. When a session is created, if we have a user id, set it there.

### Why is it needed?
This will allow easy access to all of a user's sessions, if you want to force a single user out of all their sessions you can delete all records for their user id. Currently the `modx_user_attributes` table has a `sessionid` column that stores the most recent session id, but it overwrites any previous ones. So the user keeps access to those.

### Suggested solution(s)
Adding a user id to the table should work. There may be a reason I'm unaware of making this a bad idea though, so if there is feel free to tell me to go away :D

**Comment by rthrash** [comment]
@muzzwood your request makes snese, but there are also security implications around exposing this information that unfortunatley mean this won't happen. Please see #16697

**Comment by pbowyer** [comment]
I've got to ask what those security implications are. There's needs to be a link between the session and the user for logins to work, so how does adding a `user_id` column to the `modx_sessions` table make it less secure?

I've checked https://github.com/modxcms/revolution/pull/16697 which is similar but not identical, and there isn't an explanation there.

**Comment by opengeek** [comment]
Linking a user_id and session_id is precisely the security issue. It is why it was deprecated in #16697… if someone gains access to the table, they could get the session_id for the user and then use it for access. Similarly, if the user_id was on the session table, it would have the same issue. Outside of the security implications, this would only be part of the session data if you used the default session handler in MODX—other session handlers would not have this data.

**Comment by muzzwood** [comment]
That's a pity, but makes sense. Thanks for letting me know!

**Comment by pbowyer** [comment]
My confusion is that if someone gains access to the `modx_session` table, they have both already for logged-in users:

![Image](https://github.com/user-attachments/assets/0dc89367-ae5c-4612-86f7-d4ec2aecc6b2)

We don't encrypt context tokens with a key stored outside the database (protecting against database compromise but not concurrent file system compromise) so the user_id and session_id are linked in plain text in the database?

**Comment by opengeek** [comment]
> We don't encrypt context tokens with a key stored outside the database (protecting against database compromise but not concurrent file system compromise) so the user_id and session_id are linked in plain text in the database?

You are correct, and that could be addressed, as well… though performance considerations have to be taken into account. Regardless, adding an even more direct linking is not an improvement.
